### PR TITLE
Move sigact dump/restore code into separate file

### PR DESCRIPTION
**Pull request by rakki-18**
This tries to solve the issue #336.

Changes introduced in this pull request:
- `parasite_dump_sigacts_seized`  in parasite-syscall.c is moved to sigacts-dump.c
- `prepare_sigactions` and the other functions that it calls are also moved to sigacts- dump.c
- sigacts-dump.o is added to makefiles.
- sigacts-dump.h is created and this header is included in cr-dump.c and cr-restore.c

### CODE REVIEW

**Comment by rakki-18** [comment]
can someone please review this ?

**Comment by xemul** [comment]
@rakki-18 , I've a feeling that your PR is not git-bisect safe. If this is just in moving the code from one file to another, why do we need so many patches at all? Isn't it better just to squash all the changes into a single patch?

**Comment by rakki-18** [comment]
@xemul I have squashed all the changes into one commit.

**Comment by Snorch** [comment]
@rakki-18 You've lost newlines around restore_one_sigaction also it would be nice to have them above sa_inherited and below parasite_dump_sigacts_seized. Except that functions diff looks fine.

File: criu/sigacts-dump.c [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+#include <sys/wait.h>
+#include <sys/mman.h>
+
+#include "common/config.h"
+#include "common/compiler.h"
+#include "types.h"
+#include "protobuf.h"
+#include "images/sa.pb-c.h"
+#include "images/timer.pb-c.h"
+#include "images/vma.pb-c.h"
```
Comment (Snorch) [inline criu/sigacts-dump.c:14]: Probably the old story about headers (https://github.com/checkpoint-restore/criu/pull/990#discussion_r407265715) can be continued here, why do you need vma.pb-c.h ?

Comment (rakki-18) [reply criu/sigacts-dump.c:14]: Thanks for reviewing. I will try and remove the unnecessary headers.

Comment (rakki-18) [reply criu/sigacts-dump.c]: @Snorch I have removed the unnecessary headers.

**Comment by Snorch** [comment]
>> @rakki-18 , I've a feeling that your PR is not git-bisect safe. If this is just in moving the code from one file to another, why do we need so many patches at all? Isn't it better just to squash all the changes into a single patch?

> @xemul I have squashed all the changes into one commit.

@rakki-18 again. Please think twice before you push =)

File: criu/sigacts-dump.c (hunk 1 above)
Comment (Snorch) [reply criu/sigacts-dump.c]: You do all the same mistakes again and again https://github.com/checkpoint-restore/criu/pull/990#discussion_r401401455

All external calls should have their headers, all headers for which you don't have any external call should be removed.

Comment (rakki-18) [reply criu/sigacts-dump.c]: @Snorch Oh Sorry, I will try and correct it.
The sigpending.c test is failing as the value of numsig is coming out to be one less than sent_sigs. any idea why ?

**Comment by rakki-18** [comment]
@Snorch Can you please check now ?

**Comment by rst0git** [comment]
@rakki-18 nice work!

Just a small piece of advice on your commit message:

>  Move parasite_dump_sigacts_seized and prepare_sigactions to sigacts-dump
> 
> Moved parasite_dump_sigacts_seized() to sigacts-sump.c from parasite-syscall.c.
> Moved prepare_sigactions() and the other functions it calls from cr-restore.c to cr-dump.c
> Added sigacts-dump.h. sigacts-dump.o was added to makefiles.

It would be good to have a shorter subject line and explain *why* this change is needed. I would recommend this blog post for more information: https://chris.beams.io/posts/git-commit/

**Comment by rakki-18** [comment]
@rst0git Thanks for the advice! I will try to make better commit messages from now.

**Comment by Snorch** [comment]
> https://chris.beams.io/posts/git-commit/

It's really nice, except for paragraph 3, if you do "git log --oneline" you will see that we don't capitalize in commit subject, so @rakki-18  please fix it in you commit.

**Comment by Snorch** [comment]
Moving some interface to a separate file, but still leaving some global variable `sigchld_act` which is used in both files, looks wrong for me.

More over the linker on fedora detects multiple definition on your code:

```
make -j5
...
ld: criu/sigacts-dump.o:/home/snorch/devel/criu/criu/sigacts-dump.c:63: multiple definition of `sigchld_act'; criu/cr-restore.o:(.bss+0x20): first defined here
make[2]: *** [/home/snorch/devel/criu/scripts/nmk/scripts/build.mk:164: criu/built-in.o] Error 1
make[1]: *** [criu/Makefile:76: criu/built-in.o] Error 2
make: *** [Makefile:259: criu] Error 2

```

**Comment by rakki-18** [comment]
@Snorch I have fixed the capitalisation of the commit subject.

When I had the static declarations  of  ```sigchld_act``` in both the files, it was failing the sigpending.c test as  the variable ```numsig``` in sigpending was lesser than the variable```sent_sigs``` by one.  Moving the interface to a separate file and keeping the global declarations of ```sigchld_act```  fixed the error.
any idea on how else I can solve this error ?

**Pull request closed without merging**